LCM/dsc/engine/ConfigurationManager/LocalConfigurationManager.c

```diff
@@ -50,7 +50,91 @@ MI_Boolean g_hPSInfraModuleHandleInitialized;
 
 #include "LocalConfigurationManager.h"
 
-// TODO: Temporary workaround until we can reload OMI provider registration without restarting omi.
+typedef struct _LCMTaskNode LCMTaskNode;
+
+struct _LCMTaskNode
+{
+    ThreadProc task;
+    void* params;
+    LCMTaskNode * next;
+};
+
+static LCMTaskNode * g_TaskHead = NULL;
+static pthread_mutex_t g_TaskQueueMutex;
+static int g_TaskQueueShutdown = 0;
+static int g_TaskQueueExists = 0;
+
+void AddToTaskQueue(LCMTaskNode * node)
+{
+    LCMTaskNode* current;
+    pthread_mutex_lock(&g_TaskQueueMutex);
+    if (g_TaskHead == NULL)
+    {
+	g_TaskHead = node;
+    }
+    else
+    {
+	current = g_TaskHead;
+	while (current != NULL)
```

Comment:
This is a linked list implementation to keep it simple.  It's obviously not efficient, but the number of queued tasks should never get larger than a few, so there's no need to worry about performance here.  If the number of queued tasks gets larger than a few, there's an underlying serious problem.